src/components/NavBar/EditorDesignName.tsx

```diff
@@ -36,6 +40,21 @@ const DesignName = () => {
     );
   };
 
+  const saveNameChange = () => {
+    if (newDesignName.trim().length > DESIGN_NAME_MAX_CHAR_LENGTH) {
+      setFeedbackModalOpen(true);
+      setFeedback(`The design name should less than ${DESIGN_NAME_MAX_CHAR_LENGTH} characters.`);
+    } else {
+      const errorMessage = checkDesignName(newDesignName.trim(), nameList);
+      if (errorMessage) {
+        setFeedbackModalOpen(true);
+        setFeedback(errorMessage);
+      } else {
+        dispatch(changeDesignName(newDesignName.trim()));
+      }
+    }
+  };
+
```

Comment:
The name length check seems like part of the check design name function. It seems like the length check can be moved into the checkDesignName().